tests/sentry/incidents/endpoints/test_serializers.py

```diff
@@ -373,6 +383,67 @@ def test_threshold_type(self):
         )
         self.run_fail_validation_test({"thresholdType": 50}, {"thresholdType": invalid_values})
 
+    @patch(
+        "sentry.integrations.slack.utils.get_channel_id_with_timeout",
+        return_value=("#", None, True),
+    )
+    def test_channel_timeout(self, mock_get_channel_id):
+        self.setup_slack_integration()
+        trigger = {
+            "label": "critical",
+            "alertThreshold": 200,
+            "actions": [
+                {
+                    "type": "slack",
+                    "targetIdentifier": "my-channel",
+                    "targetType": "specific",
+                    "integration": self.integration.id,
+                }
+            ],
+        }
+        base_params = self.valid_params.copy()
+        base_params.update({"triggers": [trigger]})
+        serializer = AlertRuleSerializer(context=self.context, data=base_params)
+        # error is raised during save
+        assert serializer.is_valid()
+        with self.assertRaises(ChannelLookupTimeoutError) as err:
+            serializer.save()
+        assert (
+            err.exception.message
```

Comment:
Failing in py3, you just want to do `str(err.exception)`, which I think should work fine in both.